pkg/library/status/check.go

```diff
@@ -51,12 +51,34 @@ var unrecoverablePodEventReasons = map[string]int32{
 
 var unrecoverableDeploymentConditionReasons = []string{
 	"FailedCreate",
+	"ProgressDeadlineExceeded",
```

Comment:
@amisevsk if we catch the `ProgressDeadlineExceeded` condition as I'm doing here, we should probably also specify `ProgressDeadlineSeconds` explicitly in the deployment spec (as 10 minutes is a long time to wait to see if the deployment rollout failed). Maybe `config.workspace.progressTimeout` would be a reasonable parameter to use for this?